Apps/DBMaintainer/Apps/BCProvDrugDBApp.cs

```diff
@@ -53,13 +53,16 @@ protected override string configurationName
         /// <inheritdoc/>
         public override void ProcessDownload(string sourceFolder, FileDownload downloadedFile)
         {
+            downloadedFile.ProgramTypeCodeId = Database.Constant.ProgramType.Provincial;
             string[] files = Directory.GetFiles(sourceFolder, "pddf*.csv");
             if (files.Length > 1)
             {
                 throw new ApplicationException($"The zip file contained {files.Length} CSV files, very confused.");
             }
             this.logger.LogInformation("Parsing Provincial PharmaCare file");
             List<PharmaCareDrug> pharmaCareDrugs = this.parser.ParsePharmaCareDrugFile(files[0]);
+            // inject the FileDownload into the pharmacare drugs
+            pharmaCareDrugs.ForEach(c => c.FileDownload = downloadedFile);
```

Comment:
Instead of iterating over the new entries, how about passing the fileDownload to the parser?

Reply:
I've changed this as discussed.